packages/mobile/screens/sign_up.tsx

```diff
@@ -0,0 +1,224 @@
+import * as React from "react";
+import { StyleSheet, Text, View, TextInput, TouchableOpacity, Alert, Image } from 'react-native';
+import { Formik, Field } from 'formik';
+import * as Yup from "yup";
+import { useMutation } from "@apollo/react-hooks";
+import gql from "graphql-tag";
+
+// import {CREATE_USER_MUTATION} from "@workspace-library/core";
+
+const CREATE_USER_MUTATION = gql`
+  mutation signup($username: String, $email: String!, $password: String!) {
+    signUpUser(userInput: { username: $username, email: $email, password: $password }) {
+      id
+    }
+  }
+`;
+
+const signupSchema = Yup.object().shape({
+  username: Yup.string()
+    .required("Please enter your name")
+    .min(4, "Name is too short")
+    .max(12, "Thats a long name you have there"),
+  email: Yup.string()
+    .email("Please enter a valid email")
+    .required("Please enter an email"),
+  password: Yup.string().required("Please enter the password"),
+  confirmation: Yup.string()
+    .required("Please enter the password confirmation")
+    .oneOf([Yup.ref("password")], "Password does not match")
+});
+
+const SignUp = ({ navigation }: any) => {
+  const [signUpUser, { data }] = useMutation(CREATE_USER_MUTATION);
+
+  const submition = (values: any, { setErrors }: any) => {
+    signUpUser({
+      variables: {
+        username: values.username,
+        email: values.email,
+        password: values.password
+      }
+    })
+      .then(
+        ({
+          data: {
+            signUpUser: { id }
+          }
+        }) => {
+          if (id) {
+            navigation.navigate("SignIn");
```

Comment:
As a logged user I want to be redirected to the HomeScreen, you can change this when you do the JWT functionality since you'll edit this part anyway